sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/InvalidCastToInterface.cs

```diff
@@ -175,3 +175,92 @@ public void Method<T>(T generic) where T : new()
         }
     }
 }
+
+namespace Tests.Diagnostics
+{
+    namespace CSharp8
+    {
+        public class NullCoalescenceAssignment
```

Comment:
I think it would be nice to have one more test to check the result of the coalesce assignment.
e.g. 
```
	int? i = null;
	
	i ??= null;
	var r1 = (int) i; // Noncompliant
	
	i ??= 1;
	var r2 = (int) i; // Compliant
```